basket/main.py

```diff
@@ -108,7 +147,7 @@ def _get_requirements_from_tar_archive(self, path):
         TAR-gzipped or TAR-bzipped archive.
         """
         extension = self.os.path.splitext(path)[1].split('.')[1]
-        with self.tarfile.open(path, 'r:%s' % extension) as archive:
+        with closing(self.tarfile.open(path, 'r:%s' % extension)) as archive:
```

Comment:
Both `tarfile.open()` and `zipfile.ZipFile` already act as context managers and take care of closing the files. I don't think we need to use `closing` here: did that fix a particular bug?